drive_to_gps/src/current_heading.cpp

```diff
@@ -0,0 +1,69 @@
+#include "calculate_degrees.h"
+#include <drive_to_gps/CurrentHeadingConfig.h>
+#include <drive_to_gps/heading.h>
+#include <dynamic_reconfigure/server.h>
+#include <math.h>
+#include <ros/ros.h>
+#include <sensor_msgs/NavSatFix.h>
+
+/*
+Current heading is calculated in degrees with 0 degrees at North and positive is
+clockwise
+*/
+
+ros::Publisher pub_curr_head;
+sensor_msgs::NavSatFixPtr prev_gps = NULL;
+int refresh_rate = 1; // initialize 1hz as the loop rate (can be changed)
+
+void reconfigure_callback(drive_to_gps::CurrentHeadingConfig &config,
+                          uint32_t level) {
+  ROS_INFO("Rate = %d", config.refresh_rate);
+  refresh_rate = config.refresh_rate;
+}
+
+// determine the current heading of the rover
+void determine_curr_heading(const sensor_msgs::NavSatFixConstPtr &curr_gps) {
+  // if there is no prev_gps coordinate return flag for unknown heading
+  drive_to_gps::heading msg;
+  if (prev_gps == NULL) {
+    prev_gps = sensor_msgs::NavSatFixPtr(new sensor_msgs::NavSatFix());
+    ROS_INFO("Prev GPS is NULL and curr GPS has latitude %lf and longitude %lf",
+             curr_gps->latitude, curr_gps->longitude);
+    msg.valid = false;
+    msg.degrees = 0;
+  } else {
+    ROS_INFO("Lat1 %lf, Lat2 %lf, Lon1 %lf, lon2 %lf", prev_gps->latitude,
```

Comment:
It's also probably a good idea to prefer to always use the STREAM variants of rosconsole logs. this allows you to use syntax closer to C++'s std::cout